### PR TITLE
provide lambda for writing mesh files

### PR DESCRIPTION
**Pull request by tcaduser**
*Description of changes:*

Instead of a regex, use an lambda to determine whether a field should be written to a Tecplot or VTK file.

```
devsim.write_devices(
    file="mesh2d_reduced.tec",
    type="tecplot",
    include_test=lambda x: x in ("NetDoping",),
)
```

Also remove floops/floods writer.

*Licensing*

To maintain the integrity of the project, we require contributions to be licensed under the Apache License Version 2.0. Please acknowledge this by checking the box below. By checking the box, you also acknowledge that you have the authority to license your contribution in this manner. You should also ensure that any new file you add contains "# SPDX-License-Identifier: Apache-2.0" in the header.

 - [ x] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](https://apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -5,6 +5,24 @@
...
+
+### Centos 7 End Of Life
+
+### Data output
+
+#### Reduction in data file sizes
+
+Based on a contribution by [@simbilod](https://github.com/simbilod) `devsim.write_devices` now supports reducing the file size of data files by specifying a callback function to reduce data usage.  This is through the `include_test` option.  This option is a function that takes a string and returns `True` if the model should be written.  It is only supported for the `tecplot` and `vtk` formats.
+
+An example is provided in `testing/XXXXX.py`
```
Comment (simbilod) [inline CHANGES.md:18]: should XXXXX be mesh2d.py ?

File: src/MathEval/MathEval.cc
Comment (simbilod) [inline src/MathEval/MathEval.cc:1]: It is not immediately obvious to me how this change relates to the commit message and PR. Having a separate commit with a message explaining why this is removed could be good for documentation purposes. But we can just merge this time :)

**Review by simbilod — CHANGES_REQUESTED** [review]

**Comment by simbilod** [comment]
Yes it is better to simplify to a single flag and a single share object reused across the writers to handle this. Thanks!

File: src/MathEval/MathEval.cc
Comment (tcaduser) [reply src/MathEval/MathEval.cc:1]: The format was very used.  It was supposed to be compatible this FLOOPS/FLOODS simulator output for visualization in a software named Postmini

File: CHANGES.md (hunk 1 above)
Comment (tcaduser) [reply CHANGES.md:18]: thanks, I'll clean this up at the end.

**Comment by tcaduser** [comment]
The stuff with MathEval were dead code that was never used.

**Pull request merged**